### PR TITLE
Improve Kafka consumer application

### PR DESCRIPTION
**Pull request by srikartati**
We should not assume that Kafka broker is already present and ready. 
Have a retry mechanism for consumer initialization.

### CODE REVIEW

**Comment by srikartati** [comment]
Verified on Antrea with the updated local Kafka consumer image.

File: cmd/consumer/consumer.go [hunk 1]
```diff
@@ -85,14 +90,24 @@ func run() error {
 		KafkaLogErrors:    true,
 	}
 	consumer := consumer.NewKafkaConsumer(input)
+	consumerInitError := make(chan struct{})
```
Comment (zyiou) [inline cmd/consumer/consumer.go:93]: Are we supposed to pass some error into this channel in goroutine?

Comment (srikartati) [reply cmd/consumer/consumer.go]: I was trying something else and figured it's not needed. Forgot to remove all the related code.

File: cmd/consumer/consumer.go [hunk 2]
```diff
@@ -86,9 +88,18 @@ func run() error {
 	}
 	consumer := consumer.NewKafkaConsumer(input)
 	go func() {
-		if err := consumer.InitSaramaConsumer(); err != nil {
+		err := wait.PollImmediateInfinite(500*time.Millisecond, func() (bool, error) {
+			if err := consumer.InitSaramaConsumer(); err != nil {
+				return false, nil
```
Comment (zyiou) [inline cmd/consumer/consumer.go:93]: Not sure whether it will affect condition of Poll, we may want to `return false, err` here

Comment (srikartati) [reply cmd/consumer/consumer.go:93]: If we return error, then we error out from the wait method. This behavior will be same as before.

Comment (zyiou) [reply cmd/consumer/consumer.go:93]: Got it. Thanks for clarifying

**Review by zyiou — APPROVED** [review]
LGTM

**Pull request merged**